### PR TITLE
Add ThisAssembly.Resources

### LINKED ISSUE #45: Add ThisAssembly.Resources
### Discussed in https://github.com/devlooped/ThisAssembly/discussions/34

<div type='discussions-op-text'>

<sup>Originally posted by **viceroypenguin** January 23, 2021</sup>
Proposal: Add another project for exposing `string` and `Stream` versions of embedded resources. Accessing the resources is something necessary for most source generators, but also for many other projects; you're accessing embedded resources already in the other ThisAssembly.* projects. It would be nice to be able to have compile-time named access to the embedded resources. Thoughts?</div>

## Design

Given the following items in a project:

```xml
<PropertyGroup>
   <EmbeddedResourceStringExtensions>$(EmbeddedResourceStringExtensions);.sql</EmbeddedResourceStringExtensions>
</PropertyGroup>

<ItemGroup>
    <EmbeddedResource Include="Foo.txt" />
    <EmbeddedResource Include="MyTemplate.template" Type="Text" />
    <EmbeddedResource Include="Data\Bar.json" />
    <EmbeddedResource Include="..\..\Branding\Icon.png" Link="Branding\Icon.png" Comment="Main app logo" />
    <EmbeddedResource Include="Resources.resx" />
</ItemGroup>
```

The generator would emit a class like the following:

```csharp
partial class ThisAssembly
{
  public static partial class Resources
  {
      // <summary>Resources\Foo.txt</summary>
      public static partial class Foo 
      {
        public static string Text => // return cached string version of the resource
        public static byte[] GetBytes() => // read bytes from resource
        public static Stream GetStream() => // get the resource stream
      }

      // <summary>MyTemplate.template</summary>
      public static partial class MyTemplate 
      {
        public static string Text => // return cached string version of the resource
        public static byte[] GetBytes() => // read bytes from resource
        public static Stream GetStream() => // get the resource stream
      }

      // <summary>Resources.resx</summary>
      public static partial class Resources
      {
        public static byte[] GetBytes() => // read bytes from resource
        public static Stream GetStream() => // get the resource stream
      }

      public partial class Branding 
      {
        // <summary>Main app logo</summary>
        // <remarks>Branding\Icon.png</remarks>
        public static partial class Icon
        {
          public static byte[] GetBytes() => // read bytes from resource
          public static Stream GetStream() => // get the resource stream
        }
      }

      public partial class Data 
      {
        // <summary>Data\Bar.json</summary>
        public static partial class Bar 
        {
          public static string Text => // return cached string version of the resource
          public static byte[] GetBytes() => // read bytes from resource
          public static Stream GetStream() => // get the resource stream
        }
    }
  }
}
```

From the example, note:

1. All `EmbeddedResource` items get a static class with both `GetBytes` and `GetStream` methods for retrieval
2. Text files (i.e. `.txt` and `.json`) also get a `Text` property with a cached string read from the resource
3. Only certain file extensions within `EmbeddedResources` items are defaulted to `Text` type (therefore getting a generated `Text` property). By default, this includes `.txt;.json`, but can be be extended by appending more extensions to the `EmbeddedResourceStringExtensions` property.
4. A single `EmbeddedResource` can also be annotated with the `Type="Text"` attribute too, to get the `Text` property just for that item.
5. The items' target path (that is, `%(RelativeDir)\%(Filename)%(Extension)` or `%(Link)` (for linked files) is used by default as the class `<summary />` documentation, unless a `Comment="...."` is provided in the item, in which case that becomes the `<summary />` and the former becomes the `<remarks>`.
6. The relative directory of the item is used in the class structure to nest classes and avoid name collisions (this is prevented by the solution structure itself too at that point). This is similar to the nesting done by [ThisAssembly.Constants](https://github.com/devlooped/ThisAssembly/#thisassemblyconstants)

### PR DESCRIPTION
**Pull request by viceroypenguin**
Fixes #45

### CODE REVIEW

**Comment by viceroypenguin** [comment]
@kzu NB: this is based on top of `embedded-resources`. Will be easier to read once #132 is merged and this is rebased.

**Comment by kzu** [comment]
#132 tested and merged. Rebased this one already for review

File: src/ThisAssembly.Resources/CSharp.sbntxt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {{~ if $0.Comment ~}}
+        /// {{ $0.Comment }}
+        {{~ else ~}}
+	    /// => @"{{ $0.Path }}"
+        {{~ end ~}}
+	    /// </summary>
+        public static partial class {{ $0.Name }}
+        {
+            {{~ if $0.IsText ~}}
+            private static string s_text;
```
Comment (kzu) [inline src/ThisAssembly.Resources/CSharp.sbntxt:22]: I have a serious dislike of prefixes and underscores on fields. Let's just call this `text`.

File: src/ThisAssembly.Resources/ThisAssembly.Resources.targets [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<Project>
+
+  <PropertyGroup>
+    <EmbeddedResourceStringExtensions>.txt;.cs;.sql;.json;.md;</EmbeddedResourceStringExtensions>
```
Comment (kzu) [inline src/ThisAssembly.Resources/ThisAssembly.Resources.targets:4]: In order to make this property fully customizable by users in their projects (including making it blank), it should go in a .props instead, I think.

**Comment by viceroypenguin** [comment]
Need to fix a bug in the hintname for the generated source file. Will fix in the morning.

**Review by kzu — CHANGES_REQUESTED** [review]
Very very minor changes requested 🙏. This looks *great*!

**Comment by kzu** [comment]
Build failed with: `CSC : warning CS8785: Generator 'ResourcesGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'The hintName 'Content/Styles/Custom.css.g.cs' contains an invalid character '/' at position 7. (Parameter 'hintName')' `.

Makes sense. You'll just need to replace the path separator with a dot (I think that reads best, instead of an underscore).

File: src/ThisAssembly.Resources/ThisAssembly.Resources.targets (hunk 2 above)
Comment (viceroypenguin) [reply src/ThisAssembly.Resources/ThisAssembly.Resources.targets:4]: Makes sense - gonna be honest; didn't know there was a difference between the two until you commented and I did some research. :)

File: src/ThisAssembly.Resources/CSharp.sbntxt (hunk 1 above)
Comment (viceroypenguin) [reply src/ThisAssembly.Resources/CSharp.sbntxt:22]: Personal preferences - even in C#, the bane of existence. :) 

Another difference between us... private methods should have `private` marked on them, but that's an error in your repo. :)

**Comment by viceroypenguin** [comment]
> Build failed with: `CSC : warning CS8785: Generator 'ResourcesGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentException' with message 'The hintName 'Content/Styles/Custom.css.g.cs' contains an invalid character '/' at position 7. (Parameter 'hintName')' `.
> 
> Makes sense. You'll just need to replace the path separator with a dot (I think that reads best, instead of an underscore).

Yeah - I did it the windows way cause that's what I tested on my machine; forgot about linux machines. Easy fix.

**Comment by viceroypenguin** [comment]
NB: I made reference to an image for ThisAssembly.Resources, but my VS system is so heavily color customized that I can't make the image. You mind adding that?

**Comment by viceroypenguin** [comment]
I loaded up a few VMs to test the build error - I can confirm it's an issue with the latest 7.0-rc image, but not with 6.0 RTM or 7.0 RTM. What would you like to do with this info?

**Comment by viceroypenguin** [comment]
I tested the build on a linux machine and am not able to replicate on 6.0 RTM:

```
>> docker run -it --rm mcr.microsoft.com/dotnet/sdk:6.0-alpine07:49:11 
/ # git clone https://github.com/viceroypenguin/ThisAssembly.git
Cloning into 'ThisAssembly'...
remote: Enumerating objects: 1746, done.
remote: Counting objects: 100% (590/590), done.
remote: Compressing objects: 100% (232/232), done.
remote: Total 1746 (delta 471), reused 409 (delta 346), pack-reused 1156
Receiving objects: 100% (1746/1746), 1020.31 KiB | 4.00 MiB/s, done.
Resolving deltas: 100% (1284/1284), done.
/ # cd ThisAssembly/
/ThisAssembly # dotnet build
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
  Restored /ThisAssembly/src/ThisAssembly/ThisAssembly.csproj (in 1.61 sec).
  Restored /ThisAssembly/src/ThisAssembly.Prerequisites/ThisAssembly.Prerequisites.csproj (in 1.61 sec).
  Restored /ThisAssembly/src/ThisAssembly.Strings/ThisAssembly.Strings.csproj (in 1.69 sec).
  Restored /ThisAssembly/src/ThisAssembly.Constants/ThisAssembly.Constants.csproj (in 1.69 sec).
  Restored /ThisAssembly/src/ThisAssembly.Project/ThisAssembly.Project.csproj (in 1.69 sec).
  Restored /ThisAssembly/src/ThisAssembly.AssemblyInfo/ThisAssembly.AssemblyInfo.csproj (in 1.69 sec).
  Restored /ThisAssembly/src/ThisAssembly.Metadata/ThisAssembly.Metadata.csproj (in 1.69 sec).
  Restored /ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj (in 3.54 sec).
  ThisAssembly.Prerequisites -> /ThisAssembly/src/ThisAssembly.Prerequisites/bin/Debug/ThisAssembly.Prerequisites.dll
  Created package at /ThisAssembly/bin/ThisAssembly.Prerequisites.42.42.42.nupkg.
  ThisAssembly.Metadata -> /ThisAssembly/src/ThisAssembly.Metadata/bin/Debug/ThisAssembly.Metadata.dll
  ThisAssembly.Project -> /ThisAssembly/src/ThisAssembly.Project/bin/Debug/ThisAssembly.Project.dll
  ThisAssembly.Constants -> /ThisAssembly/src/ThisAssembly.Constants/bin/Debug/ThisAssembly.Constants.dll
  ThisAssembly.AssemblyInfo -> /ThisAssembly/src/ThisAssembly.AssemblyInfo/bin/Debug/ThisAssembly.AssemblyInfo.dll
  ThisAssembly.Strings -> /ThisAssembly/src/ThisAssembly.Strings/bin/Debug/ThisAssembly.Strings.dll
  Created package at /ThisAssembly/bin/ThisAssembly.Constants.42.42.42.nupkg.
  Created package at /ThisAssembly/bin/ThisAssembly.Strings.42.42.42.nupkg.
  Created package at /ThisAssembly/bin/ThisAssembly.Metadata.42.42.42.nupkg.
  Created package at /ThisAssembly/bin/ThisAssembly.Project.42.42.42.nupkg.
  Created package at /ThisAssembly/bin/ThisAssembly.AssemblyInfo.42.42.42.nupkg.
  ThisAssembly -> /ThisAssembly/src/ThisAssembly/bin/Debug/ThisAssembly.dll
  ThisAssembly.Prerequisites -> /ThisAssembly/src/ThisAssembly.Prerequisites/bin/Debug/ThisAssembly.Prerequisites.dll
  Created package at /ThisAssembly/bin/ThisAssembly.Prerequisites.42.42.42.nupkg.
  Created package at /ThisAssembly/bin/ThisAssembly.42.42.42.nupkg.
/ThisAssembly/src/ThisAssembly.Strings/Model.cs(12,30): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/ThisAssembly.Strings/Model.cs(23,13): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/ThisAssembly.Strings/Model.cs(36,33): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/ThisAssembly.Strings/Model.cs(38,32): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/EmbeddedResource.cs(8,38): warning CS8601: Possible null reference assignment. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
  ThisAssembly.Tests -> /ThisAssembly/src/ThisAssembly.Tests/bin/Debug/ThisAssembly.Tests.dll

Build succeeded.

/ThisAssembly/src/ThisAssembly.Strings/Model.cs(12,30): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/ThisAssembly.Strings/Model.cs(23,13): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/ThisAssembly.Strings/Model.cs(36,33): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/ThisAssembly.Strings/Model.cs(38,32): warning CS8602: Dereference of a possibly null reference. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/ThisAssembly/src/EmbeddedResource.cs(8,38): warning CS8601: Possible null reference assignment. [/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
    5 Warning(s)
    0 Error(s)

Time Elapsed 00:00:14.63
```

**Comment by kzu** [comment]
I'd just replace the slashes with dots... I'm almost sure you cannot have paths in there...

**Comment by viceroypenguin** [comment]
> I'd just replace the slashes with dots... I'm almost sure you cannot have paths in there...

Correct - See ResourcesGenerator.cs:70-72. Works correctly on windows and linux as of #[c09be94](https://github.com/devlooped/ThisAssembly/pull/134/commits/c09be94f918e23ad084b68d8f1f97a6f151f20b2), according to my testing. Not sure why the ubuntu builder failed. 

```csharp
            spc.AddSource(
                $"{resourceName.Replace('\\', '.').Replace('/', '.')}.g.cs", 
                SourceText.From(output, Encoding.UTF8));
```

**Comment by kzu** [comment]
I've got a repro failure in WSL2 Ubuntu 22.04:

```
Welcome to .NET 6.0!
---------------------
SDK Version: 6.0.403

Telemetry
---------
The .NET tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate run 'dotnet dev-certs https --trust' (Windows and macOS only).
Learn about HTTPS: https://aka.ms/dotnet-https
----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.Prerequisites/ThisAssembly.Prerequisites.csproj (in 391 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly/ThisAssembly.csproj (in 391 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.AssemblyInfo/ThisAssembly.AssemblyInfo.csproj (in 399 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.Resources/ThisAssembly.Resources.csproj (in 399 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.Project/ThisAssembly.Project.csproj (in 398 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.Constants/ThisAssembly.Constants.csproj (in 399 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/ThisAssembly.Strings.csproj (in 398 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.Metadata/ThisAssembly.Metadata.csproj (in 405 ms).
  Restored /mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj (in 607 ms).
  ThisAssembly.Prerequisites -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.Prerequisites/bin/Debug/ThisAssembly.Prerequisites.dll
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.Prerequisites.42.42.42.nupkg.
  ThisAssembly.AssemblyInfo -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.AssemblyInfo/bin/Debug/ThisAssembly.AssemblyInfo.dll
  ThisAssembly.Metadata -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.Metadata/bin/Debug/ThisAssembly.Metadata.dll
  ThisAssembly.Project -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.Project/bin/Debug/ThisAssembly.Project.dll
  ThisAssembly.Strings -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/bin/Debug/ThisAssembly.Strings.dll
  ThisAssembly.Resources -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.Resources/bin/Debug/ThisAssembly.Resources.dll
  ThisAssembly.Constants -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.Constants/bin/Debug/ThisAssembly.Constants.dll
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.Constants.42.42.42.nupkg.
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.Resources.42.42.42.nupkg.
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.Metadata.42.42.42.nupkg.
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.AssemblyInfo.42.42.42.nupkg.
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.Strings.42.42.42.nupkg.
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.Project.42.42.42.nupkg.
  ThisAssembly -> /mnt/c/Code/ThisAssembly/src/ThisAssembly/bin/Debug/ThisAssembly.dll
  ThisAssembly.Prerequisites -> /mnt/c/Code/ThisAssembly/src/ThisAssembly.Prerequisites/bin/Debug/ThisAssembly.Prerequisites.dll
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.Prerequisites.42.42.42.nupkg.
  Created package at /mnt/c/Code/ThisAssembly/bin/ThisAssembly.42.42.42.nupkg.
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(12,30): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(23,13): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(32,13): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/Tests.cs(50,54): error CS0117: 'ThisAssembly.Resources' does not contain a definition for 'Content' [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/Tests.cs(54,54): error CS0117: 'ThisAssembly.Resources' does not contain a definition for 'Content' [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(45,33): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(47,32): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Resources/ThisAssembly.ResourcesGenerator/ThisAssembly.Resources.EmbeddedResource.cs(8,38): warning CS8601: Possible null reference assignment. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]

Build FAILED.

/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(12,30): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(23,13): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(32,13): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(45,33): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Strings/Model.cs(47,32): warning CS8602: Dereference of a possibly null reference. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Resources/ThisAssembly.ResourcesGenerator/ThisAssembly.Resources.EmbeddedResource.cs(8,38): warning CS8601: Possible null reference assignment. [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/Tests.cs(50,54): error CS0117: 'ThisAssembly.Resources' does not contain a definition for 'Content' [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/Tests.cs(54,54): error CS0117: 'ThisAssembly.Resources' does not contain a definition for 'Content' [/mnt/c/Code/ThisAssembly/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj]
    6 Warning(s)
    2 Error(s)
```

**Comment by kzu** [comment]
I'm pretty sure you missed checking out the right branch. I cloned clean on WSL, checked-out the right branch, and I'm getting the same failure.

Notice that in your logs, the ThisAssembly.Resources package is never built/created...

**Review by kzu — CHANGES_REQUESTED** [review]
Need to figure out why this one generator is not being run on linux (repro on WSL and hosted runner alike)

**Comment by kzu** [comment]
Ok, now it's just unit tests, should be far easier to fix 😉

**Comment by viceroypenguin** [comment]
> Ok, now it's just unit tests, should be far easier to fix 😉

Yup. Saw that, but was already in bed... 😀  Will look later today

**Comment by kzu** [comment]
How do you feel about renaming `IsText` with a `Kind="Text"` instead? Would resemble the original implementation with `Type`, which I think is more flexible should we provide additional specific support for other kinds of embedded resources...

**Comment by viceroypenguin** [comment]
Thought about that, but couldn't come up with a safe name. Updated. Thanks! :)

**Comment by kzu** [comment]
Ok, this is looking great! The final thing would be running `dotnet format` on the repo root to fix the `dotnet-format` check and it's ready for merging!! 🎉🎉

**Pull request merged**